src/main/java/net/sf/jabref/gui/actions/CleanUpAction.java

```diff
@@ -445,14 +446,12 @@ private static void doCleanUpMonth(BibtexEntry entry, NamedCompound ce) {
 
     private static void doCleanUpPageNumbers(BibtexEntry entry, NamedCompound ce) {
         String oldValue = entry.getField("pages");
-        if (oldValue == null) {
-            return;
-        }
-        String newValue = oldValue.replaceAll(" *(\\d+) *- *(\\d+) *", "$1--$2");
-        if (!oldValue.equals(newValue)) {
-            entry.setField("pages", newValue);
-            ce.addEdit(new UndoableFieldChange(entry, "pages", oldValue, newValue));
-        }
+        // run formatter
+        new PageNumbersCleanup(entry).cleanup();
+        // new value
+        String newValue = entry.getField("pages");
+        // undo action
+        ce.addEdit(new UndoableFieldChange(entry, "pages", oldValue, newValue));
```

Comment:
we should only create the undo action if anything was actually changed (oldValue != newValue).